yt/fields/field_info_container.py

```diff
@@ -363,31 +362,6 @@ def add_field(
 
         kwargs.setdefault("ds", self.ds)
 
-        if not isinstance(function, Callable):  # type: ignore [arg-type]
-            # type-checking is disabled because of https://github.com/python/mypy/issues/11071
-            # this is compatible with lambdas and functools.partial objects
-            raise TypeError(
-                f"Expected a callable object, got {function} with type {type(function)}"
-            )
```

Comment:
note that I've dropped this part completely in the process, for two reasons:
- `callable(function)` is much more idiomatic and also faster
- ... and it's also the first thing that's checked when calling `inspect.signature`, so it's redundant